processor/deltatocumulativeprocessor/reorderprocessor/internal/gmetric/metric.go

```diff
@@ -0,0 +1,93 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package gmetric // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatocumulativeprocessor/reorderprocessor/internal/gmetric"
+
+import (
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+type Metric[T any, S Slice[S, T]] interface {
+	Gauge | Sum | Expo | Hist | Summ
+	DataPoints() S
+}
+
+type Slice[Self any, E any] interface {
+	pmetric.NumberDataPointSlice | pmetric.ExponentialHistogramDataPointSlice | pmetric.HistogramDataPointSlice | pmetric.SummaryDataPointSlice
+	List[E]
+	AppendEmpty() E
+	MoveAndAppendTo(Self)
+}
+
+type (
+	Gauge pmetric.Metric
+	Sum   pmetric.Metric
+	Expo  pmetric.Metric
```

Comment:
Minor, but can we just use full names for these?

```
Gauge
Sum
ExpHistogram
Histogram
Summary
```

I don't see the benefit to shaving characters off. Sum vs Summ in particular is very confusing